solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -90,7 +107,18 @@ contract InterchainAccountRouter is Router, IInterchainAccountRouter {
     ) external returns (bytes32) {
         CallLib.Call[] memory calls = new CallLib.Call[](1);
         calls[0] = CallLib.Call({to: target, data: data});
-        return _dispatch(_destinationDomain, abi.encode(msg.sender, calls));
+        return _dispatch(_destinationDomain, calls.format(msg.sender));
+    }
+
+    function dispatch(
```

Comment:
Optional: I would personally probably just remove this version that doesn't have value as an argument

Reply:
avoiding breaking changes to the interface and gas amounts